packages/coinstac-ui/app/render/components/pipelines/pipeline.jsx

```diff
@@ -43,23 +49,49 @@ class Pipeline extends Component {
     super(props);
 
     let consortium = null;
+    let stepToDelete = null;
+    let pipeline = null;
 
+    if(props.params.pipelineId){
+      const data = ApolloClient.readQuery({ query: fetchAllPipelinesFunc });
+      pipeline = data.fetchAllPipelines.find(cons => cons.id === props.params.pipelineId);
+      delete pipeline.__typename;
+    }
+
+    if(!pipeline){
+      pipeline ={
+        name: "",
+        description: "",
+        owningConsortium: "",
+        shared: false,
+        steps: [],
+      }
+    }
+
+    //if routed from New Pipeline button on consortium page
     if (props.params.consortiumId) {
       const data = ApolloClient.readQuery({ query: fetchAllConsortiaFunc });
       consortium = data.fetchAllConsortia.find(cons => cons.id === props.params.consortiumId);
       delete consortium.__typename;
+      pipeline.owningConsortium = consortium.id;
     }
 
     this.state = {
       owner: !props.params.consortiumId || consortium.owners.indexOf(props.auth.user.id) !== -1,
-      pipeline: {
-        steps: [],
-      },
+      pipeline: pipeline,
+      consortium: consortium,
+      startingPipeline: pipeline,
     };
 
     this.addStep = this.addStep.bind(this);
     this.moveStep = this.moveStep.bind(this);
     this.updateStep = this.updateStep.bind(this);
+    this.deleteStep = this.deleteStep.bind(this);
+    this.updatePipeline = this.updatePipeline.bind(this);
+    this.open = this.open.bind(this);
```

Comment:
Could these names be a bit more descriptive? We also have accordions on the page, and I thought that's this referred to at first. `openModal()`?